webpack/JobWizard/validation.js

```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  selectTemplateInputs,
+  selectAdvancedTemplateInputs,
+} from './JobWizardSelectors';
+import { isPositiveNumber, isValidDate } from './steps/form/FormHelpers';
+import './JobWizard.scss';
+
+export const useValidation = ({ advancedValues, templateValues }) => {
+  const [valid, setValid] = useState({});
+  const templateInputs = useSelector(selectTemplateInputs);
+  const advancedTemplateInputs = useSelector(selectAdvancedTemplateInputs);
+  useEffect(() => {
+    setValid({
+      hostsAndInputs: true,
+      advanced: true,
+      schedule: true,
+    });
+    const inputValidation = (inputs, values, setInvalid) => {
+      inputs.forEach(({ name, required, value_type: valueType }) => {
+        const value = values[name];
+        if (required && !value) {
+          setInvalid();
+        }
+        if (value && valueType === 'date') {
+          if (!isValidDate(value) && !isValidDate(new Date(value))) {
+            setInvalid();
+          }
+        }
+      });
+    };
+    inputValidation(templateInputs, templateValues, () =>
+      setValid(currValid => ({ ...currValid, hostsAndInputs: false }))
+    );
+
+    inputValidation(advancedTemplateInputs, advancedValues.templateValues, () =>
+      setValid(currValid => ({ ...currValid, advanced: false }))
+    );
+    [
+      advancedValues.timeoutToKill,
+      advancedValues.concurrencyLevel,
+      advancedValues.timeSpan,
+    ].forEach(value => {
+      if (value && !isPositiveNumber(value)) {
+        setValid(currValid => ({ ...currValid, advanced: false }));
+      }
+    });
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [advancedValues, templateValues]);
```

Comment:
On initial load this causes the useEffect to be called 23 times instead of 3.
There is no need to re check validation if the values didn't change as we are validating the values.

Reply:
since the data is being based on `advancedTemplateInputs & templateInputs` if they change I guess re-validation should happen as well. does it happen because they are objects? maybe we need to `useMemo` for them? I think @amirfefer faced a similar issue with the API hook in foreman.

Reply:
They only change when the a new template is fetched, and when that happens `advancedValues` is changed every time when setting defaults.